qemu/tests/iperf_test.py

```diff
@@ -0,0 +1,176 @@
+import os
+import re
+import shutil
+import logging
+
+from aexpect import ShellCmdError
+from avocado.utils import process
+
+from virttest import data_dir
+from virttest import utils_net
+from virttest import utils_misc
+from virttest import error_context
+from virttest import utils_package
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    iperf testing with multicast/multiqueue.
+
+    1) Boot up VM
+    2) Prepare the iperf environment
+    3) Select guest or host to start iperf server/client
+    4) Execute iperf tests, analyze the result
+    5) Finish test and cleanup host environment
+
+    :param test: QEMU test object.
+    :param params: Dictionary with the test parameters.
+    :param env: Dictionary with test environment.
+    """
+
+    def iperf_compile(src_path, dst_path, session=None):
+        """Compile iperf and return its binary file path."""
+        iperf_version = params["iperf_version"]
+        iperf_source_path = dst_path + iperf_version
+        compile_cmd = params["linux_compile_cmd"] % (src_path, dst_path,
+                                                     iperf_source_path)
+        try:
+            if session:
+                logging.info("Compiling %s in guest..." % iperf_version)
+                session.cmd(compile_cmd)
+            else:
+                logging.info("Compiling %s in host..." % iperf_version)
+                process.run(compile_cmd, shell=True, verbose=False)
+        except (process.CmdError, ShellCmdError) as err_msg:
+            logging.error(err_msg)
+            test.error("Failed to compile iperf")
+        else:
+            iperf_bin_name = re.sub(r'[-2]', '', iperf_version.split('.')[0])
+            return iperf_source_path + "/src/" + iperf_bin_name
+
+    def iperf_start(iperf_path, options, session=None, catch_data=None):
+        """Start iperf session, analyze result if catch_data."""
+        iperf_cmd = iperf_path + options
+        try:
+            info_text = "Start iperf session in %s with cmd: %s"
+            if session:
+                logging.info(info_text % ("guest", iperf_cmd))
+                data_info = session.cmd_output(iperf_cmd, timeout=120)
+            else:
+                logging.info(info_text % ("host", iperf_cmd))
+                data_info = process.system_output(iperf_cmd, timeout=120,
+                                                  verbose=False).decode()
+        except Exception as err_msg:
```

Comment:
`process.CmdError, ShellCmdError` as well?

Reply:
Maybe there are `ShellTimeoutError` and `ShellProcessTerminatedError`
And if the iperf session fails to start, I think we should catch all exception and raise it.